app/gui/view/src/notification/js.rs

```diff
@@ -165,7 +234,44 @@ impl ContainerId {
 }
 
 /// Wrapper for sending arbitrary kind of toast.
-pub fn toast(message: &str, r#type: Type, options: &JsValue) -> Result<Id, JsValue> {
+pub fn toast(message: &Content, r#type: Type, options: &JsValue) -> Result<Id, JsValue> {
     let method: &str = r#type.as_ref();
     get_toast()?.send(message, method, options)
 }
+
+
+
+// ======================
+// === Error handling ===
+// ======================
+
+/// Extension trait for `Result<T, JsValue>` that provides a method for handling JavaScript errors.
+pub trait HandleJsError<T>: Sized {
+    /// Format pretty error message.
+    fn pretty_print_error(message: Option<&str>, error: &JsValue) -> String {
+        let mut ret = String::from("Error received from JavaScript.");
+        if let Some(message) = message {
+            ret.push(' ');
+            ret.push_str(message);
+        }
+        ret.push(' ');
+        ret.push_str(&pretty_print(error));
+        ret
+    }
+
+    /// Handle JS error by logging it and returning `None`.
+    fn handle_js_err(self, message: &str) -> Option<T>;
+
+    /// Handle JS error by logging it along with dynamically generated message and returning `None`.
+    fn handle_js_err_with(self, message_provider: impl FnOnce() -> String) -> Option<T> {
+        self.handle_js_err(&message_provider())
+    }
```

Comment:
WE have very similar methods in `ResultOps` in prelude.

Reply:
I am aware, these methods are in fact implemented on top of them.
These here, however, are specific to JS.